commit 8e4dc400b7659691abdd92b3fc01ef094fc8a1e7
tree 3788f4f49eadb5eece1c60140cf0a2700081dd6a
parent a0e857eeffba8976be6bdb9f4bb011c18eb2aff7
author Stefan Richter <stefanr@s5r6.in-berlin.de> 1180300445 +0200
committer Stefan Richter <stefanr@s5r6.in-berlin.de> 1184018860 +0200

    ieee1394: eth1394: revert parent device to that in 2.6.20
    
    After ieee1394 was converted away from class_device like the networking
    subsystem was already in 2.6.21, eth1394's device may point to the
    fw-host device as its parent again like in 2.6.20.
    
    This affects userspace tools which examine the sysfs representation of
    eth1394's device.
    
    Signed-off-by: Stefan Richter <stefanr@s5r6.in-berlin.de>

---
 drivers/ieee1394/eth1394.c |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- a/drivers/ieee1394/eth1394.c	2007-07-08 19:32:17.000000000 -0400
+++ b/drivers/ieee1394/eth1394.c	2007-08-27 14:01:24.000000000 -0400
@@ -599,9 +599,7 @@ static void ether1394_add_host(struct hp
 	}
 
 	SET_MODULE_OWNER(dev);
-
-	/* This used to be &host->device in Linux 2.6.20 and before. */
-	SET_NETDEV_DEV(dev, host->device.parent);
+	SET_NETDEV_DEV(dev, &host->device);
 
 	priv = netdev_priv(dev);
 	INIT_LIST_HEAD(&priv->ip_node_list);
